homework24/src/ba/bitcamp/homework24/task1/ViewComplaints.java

```diff
@@ -0,0 +1,144 @@
+import java.sql.Connection;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+
+import javax.swing.JButton;
+import javax.swing.JFrame;
+import javax.swing.JLabel;
+import javax.swing.JScrollPane;
+
+import ba.bitcamp.homework24.task1.ComplaintPrototype.Complaint;
+import ba.bitcamp.homework24.task1.SQLUtils.ConnectToDatabase;
+
+/**
+ * ViewComplaint class extends JFrame and is used to setup GUI interface for
+ * management so they can view all complaints in database easily.
+ * 
+ * @author boris
+ *
+ */
+public class ViewComplaints extends JFrame {
+	private static final long serialVersionUID = 2815863892603059630L;
+
+	private JButton update = new JButton("Update list");
+	private JLabel message = new JLabel("List of all complaints");
+	private JLabel complaintsLabel = new JLabel();
+	private JScrollPane scroll = new JScrollPane();
+
+	private Connection conn;
+	private Statement statement;
+
+	private ArrayList<Complaint> allComplaints = new ArrayList<>();
+
+	/**
+	 * Default constructor of ViewComplaint class
+	 */
+	public ViewComplaints() {
+		statement = ConnectToDatabase.connect(conn, statement);
+		initGUIPart();
+	}
+
+	/**
+	 * Method that initalizes GUI parts
+	 */
+	private void initGUIPart() {
+		setLayout(new BorderLayout());
+
+		scroll.setViewportView(complaintsLabel);
+
+		message.setHorizontalAlignment(NORMAL);
+
+		add(update, BorderLayout.SOUTH);
+		add(message, BorderLayout.NORTH);
+		add(scroll, BorderLayout.CENTER);
+
+		update.addActionListener(new ButtonAction());
+
+		setTitle("Complaints in DB");
+		setSize(900, 500);
+		setLocationRelativeTo(null);
+		setDefaultCloseOperation(DISPOSE_ON_CLOSE);
+		setVisible(true);
+	}
+
+	/**
+	 * Method that reads all complaints logged in ArrayList of complaints, build
+	 * a string and set's JLabel with all the complaints.
+	 */
+	private void updateComplaintsLabel() {
+		StringBuilder sb = new StringBuilder();
+		sb.append("<html>");
+		for (Complaint complaint : allComplaints) {
+			sb.append(complaint);
+			sb.append("<br>");
+		}
+		sb.append("</html>");
+
+		complaintsLabel.setFont(new Font("Courier", Font.PLAIN, 14));
+		complaintsLabel.setText(sb.toString());
+	}
+
+	/**
+	 * Private inner class used to add action to update button. When button is
+	 * pressed query is sent to database to select all complaints from complaint
+	 * table. Complaints are parsed, temp Complaint object is created and if not
+	 * already added, it's added to ArrayList of complaints.
+	 * <p>
+	 * If no errors are enquired, labels are set with all complaints.
+	 * 
+	 * 
+	 * @author boris
+	 *
+	 */
+	private class ButtonAction implements ActionListener {
+
+		@Override
+		public void actionPerformed(ActionEvent e) {
+			if (e.getSource() == update) {
+
+				try {
+
+					ResultSet res = statement
+							.executeQuery("select * from complaint");
+
+					while (res.next()) {
+						Integer id = res.getInt(1);
```

Comment:
Licno uvijek izbjegavam koristiti ovu logiku tj, getInt(1) jer dovoljno je da neko doda u upit jos jedan atribut i to na pocetak i sva ti logika pada u vodu :(. Koristi Named Parametre tj, getInt("ID") jer onda i ako se doda novi atribut nece doci do greske u ovom dijelu koda.